### ISSUE TITLE
Check seems to be stalling lately (GitHub API checks.create slowness)

### ISSUE THREAD

**Issue by tir38** [stilliard/github-task-list-completed#25]
I've noticed this a few times. Checking a box triggers a check and I end up in this limbo state where the check never completes

`task-list-completed Expected — Waiting for status to be reported`

Checking/unchecking/adding comments doesn't seem to retrigger anything. I can share a repo if that would help.

**Comment by stilliard** [comment]
Hi @tir38 

I've been noticing that today again as well.  
I've checked and the server is getting a number of requests hitting it but I can see Github is having some issues with degraded performance atm: https://www.githubstatus.com/ so I'm wondering if there's an issue where not all the checks are reaching us atm? 
Will keep you updated, sorry about the issue atm.

**Comment by stilliard** [comment]
**Update:** Looks like the Github issue has resolved now - https://www.githubstatus.com/incidents/fpk08rxnqjz2
You may need to tick and then untick to try the check again one more time now that it's resolved, but let me know if you're still running into this issue.

**Comment by jlai403** [comment]
The issue is still happening for me – I've tried checking/unchecking and re-added the status check to PR's.

**Comment by stilliard** [comment]
Hi @jlai403 
Good timing as I was just looking into this again as well. 
I know earlier today Github was having issues again but they are reporting it as being resolved a couple hours ago. 
I'm going to give it a little extra time as maybe they have a backlog of these requests to get through from the issue earlier.

I've checked the server the check runs on and it's running fine, no errors and I can see requests hitting it (though I purposefully don't log any details about the requests for security). 

Interestingly, for me at least, in github it's showing for one of my PR's that the check has completed but that github is for some reason still awaiting checks? If it doesn't resolve in an hour I'll report it back to Github :)
![image](https://user-images.githubusercontent.com/77454/158866065-17dc7f8c-984d-455a-992b-bb8225d5b46c.png)

**Comment by stilliard** [comment]
@jlai403 is it running faster for you now? 

My checks seem to be running faster, though I'm still seeing the "Required statuses must pass before merging" issue. 
I've reported it to Github. 

If your issue was more of the stalling or the checks taking forever, could you try again now as github appear to be working on this again and it's running the checks themselves faster.

**Comment by jlai403** [comment]
@stilliard – appreciate the quick response. I've updated our Branch Protection rules and re-added the application – seemed to resolve the issue for now.

Thanks!

**Comment by stilliard** [comment]
Awesome, glad it's working well again for you. 

I'll keep this ticket open and will report back when Github get back to me too.

**Comment by vamcs** [comment]
Hi! It's getting stuck quite often for us and GitHub doesn't seem to be having problems (at least not right now). Do you think re-adding the app can help?

Edit: not entirely sure if this happens every time, but I noticed it's stalling after a force push to a branch when rebasing it.

**Comment by stilliard** [comment]
Hi @vamcs 

Might be worth trying though normally editing the name or body of the PR will help send it over to us again. 

From my debugging, normally github send it to us quickly, but then we make several API requests back to github to check the comments, review comments & diff comments of the review, lately I've noticed these can at random times take a while for github to respond on. 

My checks seem to be running ok for me atm but could be an intermittent issue.

**Comment by stilliard** [comment]
Just as an update for the original issue, github replied last week about the ticket I'd submitted and confirmed the api issues they had at the time were likely the cause. 
Seems like their api runs a little slow sometimes but not much we can do about that, couple ideas that might work:
1. We could set timeouts on the api calls, so if they run too slow we just bail, but, that would mean we'd risk approving a PR that had outstanding issues missed which I think would be the wrong way to fix this. 
2. Currently we do the 3 api calls for PR comments, review comments & diff level comments on reviews one after another (ref https://github.com/stilliard/github-task-list-completed/blob/master/index.js#L35) we could switch to do all 3 at the same time, that should cut the time down so if each take about 3 seconds then it'd take a max of the longest 3 seconds rather than 9. Think it's worth adding a card to add this one in :)

**Comment by prtcl** [comment]
We have also been experiencing this issue. In the case of the screenshot below, this PR is approved and all tasks are checked. Removing a task or re-checking it does not seem to "kick" it. 

Is there a workaround in place? Perhaps merging the main brach back into the PR will re-check the task status?

<img width="861" alt="Screen Shot 2022-04-11 at 11 18 22 AM" src="https://user-images.githubusercontent.com/1317069/162772234-3861aedb-5650-4c31-ac8b-04af8c05bbf7.png">

**Comment by stilliard** [comment]
Hi @prtcl 
Interesting, thanks for reporting. 

I've not had this since the last issues a couple weeks back, looks like github's not reporting any issues. 

> Apr 11 18:30:16: PR ID: 6725
> Apr 11 18:30:16: PR ID: 2392
> Apr 11 18:30:16: 17:30:16.522Z  INFO http: POST / 200 - 0.57 ms
> Apr 11 18:30:16: PR ID: 6725 - COMMENTS
> Apr 11 18:30:16: PR ID: 2392 - COMMENTS
> Apr 11 18:30:16: PR ID: 1316 - REVIEW COMMENTS
> Apr 11 18:30:16: 17:30:16.878Z  INFO http: POST / 200 - 0.59 ms
> Apr 11 18:30:16: 17:30:16.907Z  INFO http: POST / 200 - 0.46 ms
> Apr 11 18:30:16: PR ID: 6725 - REVIEW COMMENTS
> Apr 11 18:30:16: 17:30:16.989Z  INFO http: POST / 200 - 0.52 ms
> Apr 11 18:30:17: PR ID: 2392 - REVIEW COMMENTS
> Apr 11 18:30:17: 17:30:17.082Z  INFO http: POST / 200 - 0.48 ms
> Apr 11 18:30:17: PR ID: 6725 - DIFF COMMENTS
> Apr 11 18:30:17: PR ID: 6725  - DONE
> Apr 11 18:30:17: 17:30:17.195Z  INFO http: POST / 200 - 0.46 ms
> Apr 11 18:30:17: PR ID: 2392 - DIFF COMMENTS
> Apr 11 18:30:17: PR ID: 2392  - DONE

Here's some output from the log showing a couple PR's coming in and completing the checks in a couple seconds, watching them atm they seem to be all running ok.
*Process for these logs atm is they start with "PR ID: X" then logs at each API lookup for comments, review comments, diff comments and then done.*


Can you tell me the pull request ID/number for the one that's stuck atm please? 

I'm now logging the ID as that gives away nothing about the repo etc. as it's an incremental ID for each repo.
If you can provide me the ID I can check if it's showing as received and completed our side or if it's getting stuck before sending back.

Thanks.

**Comment by adar-h-healthy** [comment]
@stilliard It's happening to me as well now. Do you want me to open a new issue? Thanks!

**Comment by stilliard** [comment]
Hi @adar-h-healthy 
Ah sorry about this, up to you on creating a new issue but fine to keep it on this one too. 
Could you provide me an example Pull request ID that's not completing for you atm? I'll check my logs to see if it's getting stuck or if it's something with github being slow to update again.
Thanks.

**Comment by dave-yotta** [comment]
I get this a lot too.  Is there any way to force the check to re-evaluate and/or see the status of the api calls?  I sit there unchecky checky and I don't know if it's having any affect :D

**Comment by stilliard** [comment]
Hi @dave-yotta 
Sorry you're running into this as well. 
If you can provide me the ID of the Pull request from the url (e.g. ID 20 if the url is `https://github.com/stilliard/github-task-list-completed/pull/20`) I can check the logs on the server.

**Comment by dave-yotta** [comment]
I was waiting on #393 and #1833 but merged in the end without them.  Didn't realise this was an external service, makes sense it would get slow if you're trying to scale with the whole of github.

**Comment by stilliard** [comment]
Thanks for sending those over. 
Yep external service but what's weird is the server is fine, barely using a quarter of it's resources though today. 

Here's the logs for those ID's from the last 3 hours: (UK timezone)

```
May 11 14:42:32: PR ID: 1833
May 11 14:42:32: PR ID: 1833 - COMMENTS
May 11 14:42:33: PR ID: 1833 - REVIEW COMMENTS
May 11 14:42:33: PR ID: 1833 - DIFF COMMENTS
May 11 14:42:33: PR ID: 1833  - DONE
May 11 14:42:34: PR ID: 1833
May 11 14:42:35: PR ID: 1833 - COMMENTS
May 11 14:42:35: PR ID: 1833 - REVIEW COMMENTS
May 11 14:42:35: PR ID: 1833 - DIFF COMMENTS
May 11 14:42:35: PR ID: 1833  - DONE
May 11 14:43:12: PR ID: 1833
May 11 14:43:12: PR ID: 1833 - COMMENTS
May 11 14:43:13: PR ID: 1833 - REVIEW COMMENTS
May 11 14:43:13: PR ID: 1833 - DIFF COMMENTS
May 11 14:43:13: PR ID: 1833  - DONE

May 11 14:42:47: PR ID: 393
May 11 14:42:47: PR ID: 393 - COMMENTS
May 11 14:42:47: PR ID: 393 - REVIEW COMMENTS
May 11 14:42:48: PR ID: 393 - DIFF COMMENTS
May 11 14:42:48: PR ID: 393  - DONE
```

Looks like both were sent to us, we checked the PR and it's comments (using the 3 api endpoints for comments) and then the done call is when we sent back to github to say if it's all ok or not. 

No warnings or errors in the logs around these times either. 

I think I need to raise another ticket with Github about this as for I've seen this a number of times now where the checks are hitting the server fine, checks running quick and we respond back to say it's done fast too but then no update on the Github UI itself. 
I'm not sure if they just aren't receiving the responses properly sometimes or they queue them and for some checks they are prioritised higher than ours maybe? Will contact them and find out what's up.

**Comment by dave-yotta** [comment]
Ok yeah - I gave it probably 30-60 seconds and some hard refreshes etc, but didn't see an update.  

the "waiting for checks" time is usually very long like:

```
Task list completed / task-list-completed
Started 1h 25m 19s ago
```

(since last click on a checkbox?) but I guess that's just how it's set up, instead of just running for a couple mins each time a checkbox is ticked.

**Comment by stilliard** [comment]
The waiting for checks time is a funny one, essentially when you submit a PR or comment on one, it sends us a check. 
We then respond with either it's all ok (completed), or waiting (in_progress) if there's any checklists to complete. 
This actually normally takes less than a second.
The reason it shows as waiting is because of the status we provide them rather than our app actually waiting. 
When the comment is updated / checklist task completed, github sends us another request, we re-check and respond again generally in a second as well. 
Our app isn't actually waiting, it's just the status we use, here's the code for how this works: https://github.com/stilliard/github-task-list-completed/blob/master/index.js#L72
When creating it, there was a choice between sending back "in_progress" as the status or "completed" but with a "failure" concusion rather than "success" for PR's that have tasks still to do. 
I chose "in_progress" as this way it doesn't show as the red cross and it's the same as other checks such as the WIP check: https://github.com/wip/app/blob/152a3d261eacfe487db7661b390b1bb718e072a0/lib/free/set-status.js

What's odd is it seems like github's either not getting these back all the time, though from our side they appear to be sending to them fine. Or github are queuing up these responses and not getting to ours quickly to update the UI.

**Comment by stilliard** [comment]
Hmm, comparing ours to the WIP check which works similarly.

They have a line about setting the head_branch to blank (https://github.com/wip/app/blob/152a3d261eacfe487db7661b390b1bb718e072a0/lib/free/set-status.js#L11)
they set the started_at to be the time it completes rather than when it started (https://github.com/wip/app/blob/152a3d261eacfe487db7661b390b1bb718e072a0/lib/free/set-status.js#L14) 
and they also have a request retries bit (https://github.com/wip/app/blob/152a3d261eacfe487db7661b390b1bb718e072a0/lib/free/set-status.js#L28).

I'll investigate these and see if maybe we need to modify our request back to help fix these issues.

**Comment by stilliard** [comment]
I've made the changes, still seems to be a little slow but my test request completed on the server in under 1second but github took over 2 and a half minutes to update the UI. 

They have now however put a notice on their status page about issues with PR's so it's possible the delays we've seen again today are to do with this:
https://www.githubstatus.com/

**Comment by stilliard** [comment]
Hey, 
So Github has updated their status page and the PR issues appear resolved which is awesome! 

To help I've made a couple tweaks to retry requests based on how the WIP check works. 
I've also upgraded the repo to the latest probot release so that any other upstream fixes to requests from github are applied as well to be safe. 

Hopefully the combination works well, testing on a couple repo's the updates run on the server side in less than a second still but on the github UI they are updating in a few seconds too. 

If there's any further issues with checks not completing then please check to see if github is having issues over on https://www.githubstatus.com/ and if nothing is reported please either comment here or start a new issue if needed. :octocat:

**Comment by adar-h-healthy** [comment]
Thanks for your work, @stilliard.

**Comment by stilliard** [comment]
No worries, glad it seems resolved now :)

**Comment by TicClick** [comment]
we at osu-wiki still run into this every few weeks or so, and the check takes ~15-20m to complete, or even more:

- <https://github.com/ppy/osu-wiki/pull/7379>
- <https://github.com/ppy/osu-wiki/pull/7382>

**Comment by stilliard** [comment]
Hi @TicClick 
Sorry for not replying sooner. 
I've done some checks and looks like everything's completing our side in on average 1-2 seconds, but seems like github just isn't updating their side very quickly. 
Not sure what we can do to help with that other than report it to Github again. 
Is this still happening for you today?

**Comment by stilliard** [comment]
@TicClick 

Here's a full log from our app for that first PR 7379 for example:

In the logs output you'll see each check start with "Request received", then 3 api lookups for comments on the PR, and finally "Complete and sending back to github" when we send back to Github to say it's finished:

> May 31 14:12:48 INFO (probot): PR #7379: Request received
> May 31 14:12:49 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 14:12:49 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 14:12:50 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 14:12:50 INFO (probot): PR #7379: Complete and sending back to github
> May 31 14:17:03 INFO (probot): PR #7379: Request received
> May 31 14:17:03 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 14:17:03 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 14:17:04 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 14:17:04 INFO (probot): PR #7379: Complete and sending back to github
> May 31 14:17:38 INFO (probot): PR #7379: Request received
> May 31 14:17:38 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 14:17:39 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 14:17:39 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 14:17:39 INFO (probot): PR #7379: Complete and sending back to github
> May 31 15:51:46 INFO (probot): PR #7379: Request received
> May 31 15:51:46 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 15:51:46 INFO (probot): PR #7379: Request received
> May 31 15:51:46 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 15:51:47 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 15:51:47 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 15:51:47 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 15:51:47 INFO (probot): PR #7379: Complete and sending back to github
> May 31 15:51:48 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 15:51:48 INFO (probot): PR #7379: Complete and sending back to github
> May 31 15:55:25 INFO (probot): PR #7379: Request received
> May 31 15:55:26 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 15:55:27 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 15:55:27 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 15:55:27 INFO (probot): PR #7379: Complete and sending back to github
> May 31 15:56:14 INFO (probot): PR #7379: Request received
> May 31 15:56:14 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 15:56:15 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 15:56:16 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 15:56:16 INFO (probot): PR #7379: Complete and sending back to github
> May 31 15:59:40 INFO (probot): PR #7379: Request received
> May 31 15:59:41 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 15:59:42 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 15:59:42 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 15:59:42 INFO (probot): PR #7379: Complete and sending back to github
> May 31 16:37:28 INFO (probot): PR #7379: Request received
> May 31 16:37:29 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 16:37:29 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 16:37:29 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 16:37:29 INFO (probot): PR #7379: Complete and sending back to github
> May 31 16:51:42 INFO (probot): PR #7379: Request received
> May 31 16:51:42 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 16:51:43 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 16:51:43 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 16:51:43 INFO (probot): PR #7379: Complete and sending back to github
> May 31 16:51:44 INFO (probot): PR #7379: Request received
> May 31 16:51:44 INFO (probot): PR #7379: Main comments api lookup complete
> May 31 16:51:45 INFO (probot): PR #7379: Review comments api lookup complete
> May 31 16:51:45 INFO (probot): PR #7379: Diff comments api lookup complete
> May 31 16:51:45 INFO (probot): PR #7379: Complete and sending back to github

Thanks for providing those PR links/IDs, that's at least helped us double check the app is running the checks themselves fast, just something delaying github either receiving our check response, or them queuing it up and taking a while to update the UI?

**Comment by TicClick** [comment]
hi, it happens irregularly (didn't occur today as far as I remember). wouldn't be surprised if it's github's events being ostensibly -- and silently, without the status page update -- slow. thanks for the investigation.

**Comment by stilliard** [comment]
No problem, if it happens again let me know with a PR link again & I'll try to report it quickly to them, maybe GitHub support can share some light on what's happening. 
Glad it's ok again atm though.

**Comment by namse** [comment]
I have a trouble too. 
PR: https://github.com/NamseEnt/namseent/pull/228

I made this pr before add task-list-completed. other new PR is ok like https://github.com/NamseEnt/namseent/pull/231, but strangely 228 has problem.

**Comment by stilliard** [comment]
Hi @namse 

Hmm it looks like github sent us it and we sent the response back to them:

Jun 24 14:38:51 INFO (probot): PR #228: Request received
Jun 24 14:38:52 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 14:38:52 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 14:38:52 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 14:38:52 INFO (probot): PR #228: Complete and sending back to github
Jun 24 14:49:03 INFO (probot): PR #228: Request received
Jun 24 14:49:03 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 14:49:04 INFO (probot): PR #228: Request received
Jun 24 14:49:04 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 14:49:04 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 14:49:04 INFO (probot): PR #228: Complete and sending back to github
Jun 24 14:49:04 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 14:49:05 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 14:49:05 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 14:49:05 INFO (probot): PR #228: Complete and sending back to github
Jun 24 15:41:32 INFO (probot): PR #228: Request received
Jun 24 15:41:33 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 15:41:33 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 15:41:34 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 15:41:34 INFO (probot): PR #228: Complete and sending back to github
Jun 24 15:51:48 INFO (probot): PR #228: Request received
Jun 24 15:51:49 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 15:51:50 INFO (probot): PR #228: Request received
Jun 24 15:51:52 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 15:51:53 INFO (probot): PR #228: Main comments api lookup complete
Jun 24 15:51:54 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 15:51:54 INFO (probot): PR #228: Complete and sending back to github
Jun 24 15:51:55 INFO (probot): PR #228: Review comments api lookup complete
Jun 24 15:51:56 INFO (probot): PR #228: Diff comments api lookup complete
Jun 24 15:51:56 INFO (probot): PR #228: Complete and sending back to github

Sounds like github may be having a little issue again, maybe try updating something on the PR to have it try the check again?

**Comment by namse** [comment]
@stilliard Oh, I put a comment on PR and it turns OK! Thank you :)

**Comment by stilliard** [comment]
Awesome, glad that helped :+1:

**Comment by namse** [comment]
@stilliard It seems blocked once again. could you check it?
pull request: https://github.com/NamseEnt/namseent/pull/323

**Comment by stilliard** [comment]
Hi @namse 

Hmm, looks ok atm for me but occasionally Github can still be slow.
Maybe give it a little extra time for Github to realise we've sent it back to them lol

Here's the log from our side for the last hour for that PR ID:

Sep 06 14:50:31 INFO (probot): PR #323: Request received
Sep 06 14:50:31 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 14:50:31 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 14:50:32 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 14:50:32 INFO (probot): PR #323: Complete and sending back to github
Sep 06 14:54:34 INFO (probot): PR #323: Request received
Sep 06 14:54:34 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 14:54:34 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 14:54:35 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 14:54:35 INFO (probot): PR #323: Complete and sending back to github
Sep 06 15:00:53 INFO (probot): PR #323: Request received
Sep 06 15:00:53 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 15:00:53 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 15:00:54 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 15:00:54 INFO (probot): PR #323: Complete and sending back to github
Sep 06 15:01:02 INFO (probot): PR #323: Request received
Sep 06 15:01:02 INFO (probot): PR #323: Request received
Sep 06 15:01:02 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 15:01:02 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 15:01:02 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 15:01:03 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 15:01:03 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 15:01:03 INFO (probot): PR #323: Complete and sending back to github
Sep 06 15:01:03 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 15:01:03 INFO (probot): PR #323: Complete and sending back to github
Sep 06 15:20:56 INFO (probot): PR #323: Request received
Sep 06 15:20:57 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 15:20:57 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 15:20:58 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 15:20:58 INFO (probot): PR #323: Complete and sending back to github
Sep 06 15:21:12 INFO (probot): PR #323: Request received
Sep 06 15:21:13 INFO (probot): PR #323: Main comments api lookup complete
Sep 06 15:21:13 INFO (probot): PR #323: Review comments api lookup complete
Sep 06 15:21:13 INFO (probot): PR #323: Diff comments api lookup complete
Sep 06 15:21:13 INFO (probot): PR #323: Complete and sending back to github

**Comment by namse** [comment]
Oh...! Ok, If it happen next time I will wait for enough time to make sure. Thank you ^^

**Comment by stilliard** [comment]
No worries, sorry it's been a bit slow. 
I'd not noticed it being as slow recently but it's weird that github takes a while sometimes to see it's completed. I'll keep an eye on it and raise a ticket with them if we get more reports again.

**Comment by TicClick** [comment]
started happening today again since 16:42 CEST maybe (1h ago) and keeps going. affected pull requests if you need them for debugging purposes (displaying `task-list-completed Expected — Waiting for status to be reported` mostly):
- https://github.com/ppy/osu-wiki/pull/7992
- https://github.com/ppy/osu-wiki/pull/7901
- https://github.com/ppy/osu-wiki/pull/7919 (this one didn't update task count when I ticked one and froze with `task-list-completed In progress — 1 / 2 tasks completed` instead of 2/2) 
- https://github.com/ppy/osu-wiki/pull/7995

**Comment by stilliard** [comment]
Hey @TicClick 

Checked and looks like the first 3 have all passed now. 
4th still seems to show as waiting on github, but from our side here's it's log:

Sep 13 16:48:32: INFO (probot): PR #7995: Request received
Sep 13 16:48:32: INFO (probot): PR #7995: Main comments api lookup complete
Sep 13 16:48:32: INFO (probot): PR #7995: Review comments api lookup complete
Sep 13 16:48:33: INFO (probot): PR #7995: Diff comments api lookup complete
Sep 13 16:48:33: INFO (probot): PR #7995: Complete and sending back to github
Sep 13 16:48:42: INFO (probot): PR #7995: Request received
Sep 13 16:48:43: INFO (probot): PR #7995: Main comments api lookup complete
Sep 13 16:48:43: INFO (probot): PR #7995: Review comments api lookup complete
Sep 13 16:48:44: INFO (probot): PR #7995: Diff comments api lookup complete
Sep 13 16:48:44: INFO (probot): PR #7995: Complete and sending back to github
Sep 13 16:58:33: INFO (probot): PR #7995: Request received
Sep 13 16:58:33: INFO (probot): PR #7995: Main comments api lookup complete
Sep 13 16:58:34: INFO (probot): PR #7995: Review comments api lookup complete
Sep 13 16:58:34: INFO (probot): PR #7995: Diff comments api lookup complete
Sep 13 16:58:34: INFO (probot): PR #7995: Complete and sending back to github

**Comment by TicClick** [comment]
thanks for checking. the issue appears to be non-actionable since every time it is triggered by GitHub (might be https://www.githubstatus.com/incidents/3z36qpp2c0z1 this time) -- how about adding a short disclaimer to the README, or something like that?

**Comment by stilliard** [comment]
Ah thanks for finding that incident report from github. 
Yeah I think you're right, I would like to report the issue to Github again, usually what happens is I go to report it and then my example clears so they don't have anything they can test live with I guess. 
I'll keep an eye on it but for now I'm happy having anyone report the issue here so I can verify our side it's ok at least. 
Also could be an indicator that there's a larger issue is we get a number of reports all at once.

**Comment by dave-yotta** [comment]
Happening again now `yottaltd/alloy-engine/pull/1968` 15/17 but all are ticked.  No incidents on github - I'll ask github support 🤔

**Comment by stilliard** [comment]
Hey @dave-yotta , sorry for the delay in my reply there. 
Checked and looks like github's api requests were running slowly, here's the logs for that pull ID 1968 from earlier:

> Nov 09 14:53:37: INFO (probot): PR #1968: Request received
> Nov 09 14:54:42: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 14:55:04: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 14:55:16: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 14:55:16: INFO (probot): PR #1968: Complete and sending back to github
> Nov 09 14:59:06: INFO (probot): PR #1968: Request received
> Nov 09 14:59:08: INFO (probot): PR #1968: Request received
> Nov 09 14:59:16: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 14:59:22: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 14:59:30: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 14:59:34: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 14:59:44: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 14:59:44: INFO (probot): PR #1968: Complete and sending back to github
> Nov 09 14:59:46: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 14:59:46: INFO (probot): PR #1968: Complete and sending back to github
> Nov 09 15:00:00: INFO (probot): PR #1968: Request received
> Nov 09 15:00:02: INFO (probot): PR #1968: Request received
> Nov 09 15:00:16: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 15:00:23: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 15:00:47: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 15:01:16: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 15:01:46: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 15:01:46: INFO (probot): PR #1968: Complete and sending back to github
> Nov 09 15:02:24: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 15:02:24: INFO (probot): PR #1968: Complete and sending back to github
> Nov 09 15:07:49: INFO (probot): PR #1968: Request received
> Nov 09 15:07:51: INFO (probot): PR #1968: Main comments api lookup complete
> Nov 09 15:07:53: INFO (probot): PR #1968: Review comments api lookup complete
> Nov 09 15:07:54: INFO (probot): PR #1968: Diff comments api lookup complete
> Nov 09 15:07:54: INFO (probot): PR #1968: Complete and sending back to github

Looking through, there's over a minute there waiting for github to respond via their api for the first request back to them. Entire request seems to be under a couple minutes each though so I imagine that if their api was slow then they probably had some issues updating the responses of the checks too. 

Has this resolved for you now? 
Looks like github had an issue last night: https://www.githubstatus.com/incidents/74y4xlr09qll maybe it caused some follow up issues earlier too.

**Comment by dave-yotta** [comment]
seems slow to respond for yottaltd/alloy-reports/pull/232 (still waiting as of finishing this message - going to undo the ticks now :) )

I reported to github, there were no incidents at the time - could be they spotted it eventually.

**Comment by stilliard** [comment]
Hi @dave-yotta 
Hmm yeah looks like they are still having some slowness with the checks though the api requests are running much faster again today it looks like:

> Nov 10 09:21:02 INFO (probot): PR #232: Request received
> Nov 10 09:21:02 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 09:21:03 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 09:21:03 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 09:21:03 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 09:21:29 INFO (probot): PR #232: Request received
> Nov 10 09:21:29 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 09:21:30 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 09:21:30 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 09:21:30 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 09:21:30 INFO (probot): PR #232: Request received
> Nov 10 09:21:31 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 09:21:31 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 09:21:31 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 09:21:31 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:20:03 INFO (probot): PR #232: Request received
> Nov 10 10:20:07 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:20:10 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:20:12 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:20:12 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:20:44 INFO (probot): PR #232: Request received
> Nov 10 10:20:51 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:20:55 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:20:59 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:20:59 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:23:36 INFO (probot): PR #232: Request received
> Nov 10 10:23:40 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:23:42 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:23:44 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:23:44 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:23:48 INFO (probot): PR #232: Request received
> Nov 10 10:23:48 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:23:49 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:23:49 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:23:49 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:44:16 INFO (probot): PR #232: Request received
> Nov 10 10:44:16 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:44:16 INFO (probot): PR #232: Request received
> Nov 10 10:44:16 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:44:17 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:44:17 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:44:18 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:44:18 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:44:18 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:44:18 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:46:58 INFO (probot): PR #232: Request received
> Nov 10 10:46:58 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:46:59 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:46:59 INFO (probot): PR #232: Request received
> Nov 10 10:46:59 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:46:59 INFO (probot): PR #232: Complete and sending back to github
> Nov 10 10:46:59 INFO (probot): PR #232: Main comments api lookup complete
> Nov 10 10:47:00 INFO (probot): PR #232: Review comments api lookup complete
> Nov 10 10:47:00 INFO (probot): PR #232: Diff comments api lookup complete
> Nov 10 10:47:00 INFO (probot): PR #232: Complete and sending back to github

**Comment by stilliard** [comment]
Thanks for reporting it to github, let me know if they reply asking for anything about the checks code etc. it's all open source on here if they want to take a look too.

**Comment by dave-yotta** [comment]
Hey thanks, they are asking:

>Hi Dave,
>
>Thanks for reaching out!
>
>Sorry, the PR link you shared was private so I was not able to take a look. Generally, the Expected Îiting for status to be reported would be displayed if the status hasn't been sent by the integration/external service.
>
>Could you please ask the integrator to share with us the full output of curl -v requests (http://curl.haxx.se/) that indicates that the status update was sent to GitHub API? We're interested in seeing the full HTTP request and response objects for all API calls, including all headers and bodies (that's what the -v flag does). That would help with the investigation on our end.
>
>You can check the status of a reference using the API endpoint below:
>
>https://docs.github.com/en/rest/commits/statuses#get-the-combined-status-for-a-specific-reference

hopefully I won't have to play ping-pong >.<

**Comment by dave-yotta** [comment]
@stilliard Github are after some request/response headers:
> If we get the API request and response header sent to the PR by the integration, we have details to investigate this on our end.

Not sure what ones they mean, I've asked for clarification.  Might they mean those sent by the browser from the PR page, or ones sent to github from your integration? 🤔 This is still persistently happening for us, so should be easy to reproduce.

**Comment by dave-yotta** [comment]
Ok they are asking for the request/response headers from the request you are making to github corresponding to the log `Complete and sending back to github`, probably to help identify.  

I've just checked/unchecked on yottaltd/alloy-engine/pull/1979 and it's as usual being very slow to respond - took a minute to green and now taking time to go back after unchecking.

Will you be able to get that data?  Let me know if you need me to reproduce it again

**Comment by dave-yotta** [comment]
They also recommend checking using the https://github-debug.com/ tool?

Edit: If the webhooks are late (looks like they are?) GitHub support are asking
>  If hooks are being delivered extremely late. The GitHub App owner can share with us the X-GitHub-Delivery for such webhooks events.

**Comment by stilliard** [comment]
Hey @dave-yotta 
Sorry for not replying sooner! I've been away for about a week but back shortly. 
I'll take a look into the debug tools they've sent early next week, thanks for chasing them about it.

**Comment by dave-yotta** [comment]
@stilliard did you get a chance to look at this yet? I'm staring at another PR waiting for a green tick :D

**Comment by stilliard** [comment]
Hi @dave-yotta ah not yet sorry. 
I'll contact github about it again soon, lately i'd been seeing it getting back faster so I'd paused pushing forward on it, sorry it's playing up a bit more. 

I'm actually seeing the same on some other github checks such as WIP where our task-list-completed one shows as responded but not theirs:
![image](https://user-images.githubusercontent.com/77454/212752686-ed47b38c-6162-4a29-a0b7-23f108a67448.png)

Will keep try to get back onto this shortly.

**Comment by stilliard** [comment]
Given myself a kick to get going on this as it's been on the cards for too long :) sorry for not debugging further earlier. 

Tested with the https://github-debug.com/ site, unfortunately it's mostly just to check that we can connect to github and all 4 commands run fine as expected. 
We have no issues/errors connecting to github regularly. 

With the `X-GitHub-Delivery` header timing, we actually always get the webhook sent to us instantly, we then reply within a second generally as well but the issue is actually appears on occasion github queues these up *(I assume)* and they are on occasion slower to go through the queue of check responses their side, maybe from other high traffic they receive. 

As far as I've seen, we're not seeing any errors in the logs from the `context.octokit.checks.create()` calls where we send the check response to them so I believe we're able to send it to them fine too. 
Maybe I can improve our logs as currently we log just before we call `context.octokit.checks.create()` but then we return the create calls response, I'll modify to try logging after it's run that request as an extra confirm for that too.

**Comment by stilliard** [comment]
Ok i've added a log to show the http status response github gives us when we submit a check back to them. 
That'll help confirm if they are accepting the check as we expect or giving us something more unexpected. 

Let me know the next time you spot this and I'll try to jump on it quickly to check.

**Comment by TicClick** [comment]
it was happening just now, here: https://github.com/ppy/osu-wiki/pull/8633 -- the check had the `Expected — Waiting for status to be reported` status for quite a while. it may still have it -- I've just done a dummy PR description edit in an attempt to retrigger it, but before that it's been frozen for minutes

**Comment by stilliard** [comment]
Hey @TicClick  :wave: 
Clicked the link and it appears to have already resolved.

**Comment by stilliard** [comment]
Checking the new logs, it looks like github is sometimes taking a bit longer to receive our checks, I assume they are experiencing higher traffic for the checks or api atm. 
A few are returning in a few seconds, others a couple minutes, but then some like this one are taking about 40 minutes...

Jan 17 15:30:47: INFO (probot): PR #2749: Request received
Jan 17 15:30:47: INFO (probot): PR #2749: Main comments api lookup complete
Jan 17 15:30:47: INFO (probot): PR #2749: Review comments api lookup complete
Jan 17 15:30:48: INFO (probot): PR #2749: Diff comments api lookup complete
Jan 17 15:30:48: INFO (probot): PR #2749: Complete and sending back to GitHub
Jan 17 15:30:57: INFO (probot): PR #2749: Request received
Jan 17 15:30:58: INFO (probot): PR #2749: Main comments api lookup complete
Jan 17 15:30:58: INFO (probot): PR #2749: Review comments api lookup complete
Jan 17 15:30:59: INFO (probot): PR #2749: Diff comments api lookup complete
Jan 17 15:30:59: INFO (probot): PR #2749: Complete and sending back to GitHub
Jan 17 16:04:04: INFO (probot): PR #2749: Request received
Jan 17 16:04:04: INFO (probot): PR #2749: Main comments api lookup complete
Jan 17 16:04:04: INFO (probot): PR #2749: Review comments api lookup complete
Jan 17 16:04:05: INFO (probot): PR #2749: Diff comments api lookup complete
Jan 17 16:04:05: INFO (probot): PR #2749: Complete and sending back to GitHub
Jan 17 16:09:00: INFO (probot): PR #2749: Check response status from GitHub 201

Each of the "Request received" and lines under is a separate webhook we received instantly, but then after "Complete and sending back to GitHub" we send an api request to create the check back with github, once that's done we'd show "Check response status from GitHub 201" (or a different status if they had errors). 

No error statuses, but that final line for "Check response status from GitHub 201" is for after the first "Complete and sending back to GitHub" line back at 15:30:48. The other 2 haven't been received by GitHub yet.

**Comment by stilliard** [comment]
For ref, here's the logs for your PR @TicClick for the last couple hours worth

Jan 17 15:26:42: INFO (probot): PR #8633: Request received
Jan 17 15:26:42: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 15:26:43: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 15:26:43: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 15:26:43: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 15:27:45: INFO (probot): PR #8633: Request received
Jan 17 15:27:45: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 15:27:45: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 15:27:45: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 15:27:45: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 15:28:21: INFO (probot): PR #8633: Request received
Jan 17 15:28:22: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 15:28:22: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 15:28:22: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 15:28:22: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 15:33:57: INFO (probot): PR #8633: Request received
Jan 17 15:33:57: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 15:33:58: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 15:33:58: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 15:33:58: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 15:54:18: INFO (probot): PR #8633: Request received
Jan 17 15:54:18: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 15:54:18: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 15:54:18: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 15:54:18: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 15:57:38: INFO (probot): PR #8633: Check response status from GitHub 201
Jan 17 16:20:31: INFO (probot): PR #8633: Request received
Jan 17 16:20:31: INFO (probot): PR #8633: Main comments api lookup complete
Jan 17 16:20:32: INFO (probot): PR #8633: Review comments api lookup complete
Jan 17 16:20:32: INFO (probot): PR #8633: Diff comments api lookup complete
Jan 17 16:20:32: INFO (probot): PR #8633: Complete and sending back to GitHub
Jan 17 16:32:02: INFO (probot): PR #8633: Check response status from GitHub 201

Only 2 "Check response status from GitHub 201" logs, while 6 "Complete and sending back to GitHub" logs. 

4 are still with GitHub waiting for them to come back :smile:

**Comment by stilliard** [comment]
I guess as well, there's no guarantee those later "Check response status from GitHub 201"s are in order. 

It could be that the first webhook comes in, we pretty much instantly send it to github, but they accept it 40 minutes later. But during this time the 2nd or 3rd webhook could come in to us, we'd also send it github instantly again but maybe that later one gets accepted sooner. Giving the illusion that it's pushed the first through but only the 2nd or 3rd has. Then when the previous ones are finally read by github I assume it ignores them as it's already got later ones that are complete.

All assumptions though btw as I don't know how they handle the api requests internally.

**Comment by dave-yotta** [comment]
Github support are still asking for:

>It is difficult for me to troubleshoot with the info you shared. As I previously mentioned in your last ticket:
>
> Can they(the App owners) share with us the API request and response(including the headers) for the request sent to GitHub by their integration?
>
> This information(most importantly the X-GitHub-Request-Id) would enable me to look into our logs.

I'll drop a message here with the next PR which I check boxes on :)

**Comment by stilliard** [comment]
Hi @dave-yotta 
Thanks for contacting them anyway. 
I'll try reaching out to them directly and see where I can get.
Do you have a ticket/case ID with them I could reference please?

**Comment by dave-yotta** [comment]
👍 no problem, the ticket is #1970397.

They were also asking for the value of `X-GitHub-Delivery` from the late webhook responses which is an id they can use to trace the issue.

**Comment by seanweiss91** [comment]
Experiencing a similar issue. Interestingly, my co-worker has no problem going in and merging on his own device, but for me I can't get all of the checks to "pass". Tried checking/unchecking and refreshing.

I'm using chrome on a Mac, co-worker was on Firefox on Windows

**Comment by stilliard** [comment]
Hi @seanweiss91 
I'm seeing the same today too, checked the server and we're receiving the request from github right away and sending it back to them in often in under a second or 2, but, Github seems to be having delays again as the request we send back to them to save the check response is often taking multiple minutes. 

What can help is updating the PR quickly, like unchecking, waiting a few seconds and checking again, or updating the name by adding a full stop or similar. 
These little updates don't speed anything up our side, but they trigger another request, which we again send to github and what I've seen is github will on some requests be slow, but others faster. Meaning sometimes, that 2nd request will complete way before the first one :sweat_smile: .

Hope this helps, I'll try reporting to Github again soon if it keeps up though.

**Comment by seanweiss91** [comment]
I'll give it a try if it happens again, thank you! We really appreciate using the functionality in our team 👍

**Comment by stilliard** [comment]
Hi, just reporting this to Github again as it's been happening on and off today. 
Will report back when I hear more.

**Comment by peppy** [comment]
Here's an example of this still happening: https://github.com/ppy/osu-wiki/pull/10161

I've tried unchecking and rechecking the boxes but it doesn't seem to recover.

**Comment by stilliard** [comment]
Hi @peppy 
checked our logs for that ID for the last hour, looks like we got the request, parsed it and sent it back to gtithub but it's queued their side:
Oct 11 09:53:03 stapps npm[3389129]: INFO (probot): PR #10161: Request received
Oct 11 09:53:03 stapps npm[3389129]: INFO (probot): PR #10161: Main comments api lookup complete
Oct 11 09:53:04 stapps npm[3389129]: INFO (probot): PR #10161: Review comments api lookup complete
Oct 11 09:53:04 stapps npm[3389129]: INFO (probot): PR #10161: Diff comments api lookup complete
Oct 11 09:53:04 stapps npm[3389129]: INFO (probot): PR #10161: Complete and sending back to GitHub
Oct 11 09:58:33 stapps npm[3389129]: INFO (probot): PR #10161: Request received

**Comment by stilliard** [comment]
I have a change I need to make my side to help with reporting this to github, they want me to add additional logs for the responses. As far as we can see it's an issue getting through to github but often by the time their support replies it's resolved so they need some ids from the response headers from their requests to help debug. 

I'll see what I can do to get those in later today.

**Comment by peppy** [comment]
Thanks for checking on it.

The weird part seems to be that the *previous* run is stuck in a [running state](https://github.com/ppy/osu-wiki/runs/17551254578). I wonder if that is inhibiting the re-run from working.

**Comment by stilliard** [comment]
Np, sorry it's still causing issues. 

Ah yes the checks stay in the `in_progress` state if there's checklists waiting to be completed rather than saying it failed. 
It doesn't actually stall or hold up our side as in progress, instead we just tell github it's in progress and we cleanup and go on to the next request, each normally completing in under a second but we do end up waiting when the requests back to github take a while to complete. 
The `in_progress` state pattern rather than failing we borrowed from the [WIP app](https://github.com/apps/wip) so that any other checks that fail show as a full fail and you can tell the difference between actual fails vs missing tasks as they are still in progress. 

My hope is once we have the new logs github can see why sometimes they end up queuing their side.

**Comment by peppy** [comment]
If it helps, here's another stalled one with no task list in sight: https://github.com/ppy/osu/pull/25101

**Comment by stilliard** [comment]
I've pushed a new update which includes better logging for Github to help debug this issue (as well as a couple extra features for optional and skippable tasks, see readme). 

Please let me know if you see any further issues and what your repo & pr ID is, this should hopefully help us track down the cause.

**Comment by peppy** [comment]
Will keep an eye out, thanks.

**Comment by stilliard** [comment]
Since adding this I've seen a few requests running slow today so opened a new ticket with github and reported over to them with the full logs. 
Will update here when I here back.

**Comment by stilliard** [comment]
GitHub support replied to say they haven't had chance to look into it yet but will hopefully follow up with it next week. 
I sent another example over to their support as well with the X-GitHub-Request-Id. 

Also I've changed the timeouts & retries to hopefully help resolve this as well:
https://github.com/stilliard/github-task-list-completed/commit/91c6a6bcf9003ede824e2e2d9bdaccf24b6ee635

**Comment by Walavouchey** [comment]
another stalled pr https://github.com/ppy/osu-wiki/pull/10269 (took 14 minutes)

**Comment by stilliard** [comment]
Hi @Walavouchey 

No reply back from Github yet sorry. 

Here's the logs for your check:

> Oct 23 10:00:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Request received [Context: 94b4d430-7182-11ee-84b1-1bc651612b8d]
> Oct 23 10:00:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Main comments api lookup complete
> Oct 23 10:00:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Review comments api lookup complete
> Oct 23 10:00:17 INFO (event): PR Ephemeralis/osu-wiki#10269: Diff comments api lookup complete
> Oct 23 10:00:17 INFO (event): PR Ephemeralis/osu-wiki#10269: Complete and sending back to GitHub
> Oct 23 10:01:15 INFO (event): PR Ephemeralis/osu-wiki#10269: Request received [Context: b8385210-7182-11ee-9c39-2b5478fa76dd]
> Oct 23 10:01:15 INFO (event): PR Ephemeralis/osu-wiki#10269: Main comments api lookup complete
> Oct 23 10:01:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Review comments api lookup complete
> Oct 23 10:01:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Diff comments api lookup complete
> Oct 23 10:01:16 INFO (event): PR Ephemeralis/osu-wiki#10269: Complete and sending back to GitHub

all logs show it was good and we sent to github, but no reply from them yet.

**Comment by stilliard** [comment]
Until I hit send on that comment aha, good timing from github. 

> Oct 23 10:15:20 INFO (event): PR Ephemeralis/osu-wiki#10269: Check response status from GitHub 201 [X-GitHub-Request-Id: CC4A:C28E:30D6393:316D092:653639A7]
> Oct 23 10:16:13 INFO (event): PR Ephemeralis/osu-wiki#10269: Check response status from GitHub 201 [X-GitHub-Request-Id: 8662:1D69:3245177:32DA283:653639DC]

**Comment by Walavouchey** [comment]
https://github.com/ppy/osu-wiki/pull/10268 stuck for 40 minutes atm

**Comment by stilliard** [comment]
No reply from github yet, will update when I hear back from them.

One thing to try, make a quick change to the pr title or body, or untick then re-tick. 
In cases where it's been stuck a long time this can mean the following request completes before the original. 
Hope this helps

**Comment by TicClick** [comment]
yup, we do that often hoping things will improve, but it's still a game of
chance — more often than not it still takes dozens of minutes for edits to
do something, if anything at all. recent example:
https://github.com/ppy/osu-wiki/pull/10194

**Comment by stilliard** [comment]
Yeah it can be frustrating I understand, unfortunately without a reply from GitHub on it there's not much we can do as it appears to stall their side receiving in the check create call.

**Comment by stilliard** [comment]
Another option btw is you could try self hosting it. 
Github may be just as slow for you with the check, or it might not. My thought is if they are seeing too much traffic from our IP but if you try hosting this yourself you might find the checks get through. 
Would be interesting if that works as it confirms that suspicion and maybe then the solution is to use multiple servers for this rather than the single server it's on now. 

Or if you want, there's a github action based flow here: https://github.com/Shopify/task-list-checker/
It works similarly but through githubs actions instead of a separate check, though I've not personally tried it. 

Will report back when I hear anything from github still though.

----

If anyone has a contact at GitHub and wants to talk to them about this, our ticket is #2383169.
& not sure if it'd help but If anyone else wants to open a ticket with Github, reference the above ticket & let me know your PR repo & ID and I'll provide the logs to send to them :+1:

**Comment by stilliard** [comment]
I have a plan, off the idea about it possibly being traffic from our ip to github backing up. 
If we don't hear back from github soon, I'll find some time to set up a number of low cost droplets on DO with haproxy to balance between them to see if multiple lower cost servers through different ips helps :balance_scale:

**Comment by stilliard** [comment]
Had some time tonight to play around with the new idea. 

I've set up a new load balancer and a couple nodes for now (may add more as needed) dedicated to this app. 

These are now in San Francisco, which may perform better than the uk location previously on DO. 
That said, just doing some searching around, it might be that githubs servers are in Virginia so moving the servers to the DO new york datacenter may actually be better but I can do that another night. 

Monitoring it atm and it seems to be running fine however this time of day is often has less traffic for the service anyway, the real test will be tomorrow onwards. 

Let me know if you notice any issues, also worst case I can switch it back to the old setup with a single url setting change in github just in case as well.

**Comment by stilliard** [comment]
**Update since moving the servers:**
All running well so far, moving to the US & splitting to multiple servers does seem to be helping at least so far. 
It's also equally possible GitHub didn't have any slowness in the API today :sweat_smile: 
Will keep a close eye on things.

**Comment by dave-yotta** [comment]
Anecdotally, they seem to be working much better now.  I had a moment of surprise when a PR went green right after clicking a checkbox which never happened before.

**Comment by stilliard** [comment]
@dave-yotta 
Yeah it's been working without any issue reports for the last week. 
Monitoring hasn't spotted any spikes or anything either. 
Leaving this open for another week to be sure but think the move has definitely helped :)

**Comment by stilliard** [comment]
Been a couple weeks and not had any further issues. 
I think we're good to close this one off again. 
If anyone runs into the same issue feel free to reply or open a new ticket any time. 
Thanks!

**Comment by peppy** [comment]
I just noticed this one today, could you double-check? https://github.com/ppy/osu-queue-score-statistics/pull/164

**Comment by stilliard** [comment]
@peppy thanks for reporting. 

This one is interesting, I don't see it at all in the log atm. I wonder if Githubs not sent it yet for some reason?

**Comment by stilliard** [comment]
Nothing reported on the github status atm. 
Could you try editing it to see if that pushes it over? 
We have no entries of "osu-queue-score-statistics" in the logs for last couple hours on the servers but I've tested some others and they appear fine. Not sure why Github's been a little slow with sending that one?

**Comment by peppy** [comment]
Editing did help in this case.

**Comment by stilliard** [comment]
Cool, checked the logs and I can see 2 updates just came in for your repo.
Treating as a one off as unlike previous issues, Github didn't even send the request to us until the edit. 
Thanks for reporting though. Will continue to keep an eye on it :+1:
